src/common/runtime/src/repeated_task.rs

```diff
@@ -75,16 +149,19 @@ impl<E> Drop for RepeatedTask<E> {
 }
 
 impl<E: ErrorExt + 'static> RepeatedTask<E> {
-    pub fn new(interval: Duration, task_fn: BoxedTaskFunction<E>) -> Self {
+    pub fn new<I: Into<Box<dyn IntervalGenerator>>>(
```

Comment:
I think this makes things complicated. In most scenarios, what we need is an initial delay. Now our initial delay is always `interval`. We can add an argument to `new()` or provide another method to specify the initial delay:
```rust
pub fn with_initial_delay(initial_delay: Duration, interval: Duration, task_f: BoxedTaskFunction<E>) -> Self {}
```

Then we can call the task_fn first. By default, `initial_delay` is None.
```rust
loop {
    let sleep_time = initial_delay.take().unwrap_or(interval);
    tokio::select! {}
}
```